src/overlap_pmmg.c

```diff
@@ -0,0 +1,736 @@
+        if ( !(ip == ip_out) ) continue;
+
+        /* Search into dataPBDY_AlreadyAdded to seen if it exists already */
+        for (r=0; r < ndataPBDY_added+1; r++) {
+          if ( (color_out == dataPBDY_AlreadyAdded[5*r+1]) && (color_ter == dataPBDY_AlreadyAdded[5*r])) {
+            if ( k == dataPBDY_AlreadyAdded[5*r+2] ) {
+              ip_in = dataPBDY_AlreadyAdded[5*r+3];
+              duplicated_point =1;
+              break;
+            }
+          }
+          if ( (color_out == dataPBDY_AlreadyAdded[5*r]) && (ip_out == dataPBDY_AlreadyAdded[5*r+4])) {
+              ip_in = dataPBDY_AlreadyAdded[5*r+3];
+              duplicated_point =1;
+              break;
+          }
+        }
+
+        /* If this point has not been added yet, then add it to the mesh */
+        if (duplicated_point==0) {
+          if ( color_ter != color_in ) {
+            ip_in = MMG3D_newPt(mesh,coord,tag,1);
+            mesh->point[ip_in].ref = ref; // Add ref
+            mesh->point[ip_in].xp  = 0;   // Assign 0 to xp
+            ls->m[ip_in]           = ls_val;
+          }
+        }
+
+        /* Update the hash tables to know correspondance of local index on color_in and color_out */
+        hash_out2in[ip_out] = ip_in;  // From index on color_out, I found index on color_in
+        hash_in2out[ip_in]  = ip_out; // From index on color_in,  I found index on color_out
+
+        /* Update dataPBDY_AlreadyAdded data necessary to identify which MG_PARBDY points have already been treated */
+        dataPBDY_AlreadyAdded[5*ndataPBDY_added]  = color_out;
+        dataPBDY_AlreadyAdded[5*ndataPBDY_added+1]= color_ter;
+        dataPBDY_AlreadyAdded[5*ndataPBDY_added+2]= k;
+        dataPBDY_AlreadyAdded[5*ndataPBDY_added+3]= ip_in;
+        dataPBDY_AlreadyAdded[5*ndataPBDY_added+4]= ip_out;
+        ndataPBDY_added++;
+      }
+    }
+
+    /** STEP 9 - Add all the other nodes from Pcolor_out to:
+     *     a) the local mesh on Pcolor_in
+     *     b) the overlap hash tables (hash_out2in and hash_in2out) **/
+    j=0;
+
+    for (i=0; i < 4*ntTot_out2in; i++) {
+      ip_out=tetraVertices_ToRecv_outIdx[i]; // This point has the index ip_out on Pcolor_out
+
+      if (tetraVerticesSeen_ToRecv[i]==1) {
+        coord[0] = pointCoordInterior_ToRecv[3*j];
+        coord[1] = pointCoordInterior_ToRecv[3*j+1];
+        coord[2] = pointCoordInterior_ToRecv[3*j+2];
+        ref      = pointRefInterior_ToRecv[j];
+        tag      = pointTagInterior_ToRecv[j]|MG_OVERLAP; // Add the tag MG_OVERLAP to this point
+        ls_val   = lsInterior_ToRecv[j];
+        j += 1;
+
+        ip_in = MMG3D_newPt(mesh,coord,tag,1);
+        mesh->point[ip_in].ref = ref; // Assign ref
+        mesh->point[ip_in].xp  = 0;   // Assign 0 to xp
+        ls->m[ip_in]           = ls_val;
+        hash_out2in[ip_out] = ip_in;  // From index on color_out, I found index on color_in
+        hash_in2out[ip_in]  = ip_out; // From index on color_in,  I found index on color_out
```

Comment:
`hash_in2out` is not used no ?

Reply:
Indeed, `hash_in2out` is not used here - but by anticipation, it might be used by `PMMG_snpval_ls`. If finally it won't be used by `PMMG_snpval_ls`, indeed I will delete this hash table.